core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/git/GitHandler.scala

```diff
@@ -0,0 +1,97 @@
+package org.virtuslab.ideprobe.dependencies.git
+
+import org.eclipse.jgit.api.{CloneCommand, CreateBranchCommand, Git}
+import org.eclipse.jgit.lib.{Constants, ProgressMonitor, Ref}
+
+import java.net.URI
+import java.nio.file.Path
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicInteger, AtomicReference}
+import scala.util.Try
+import org.virtuslab.ideprobe.Extensions._
+
+object GitHandler {
+
+  implicit class GitRepository(repository: String) {
+    def clone(workspace: Path): Git = {
+      val command: CloneCommand = new CloneCommand()
+        .setURI(repository)
+        .setDirectory(workspace.toFile)
+        .setCloneAllBranches(true)
+        .setMirror(false)
+        .setBare(false)
+        .setProgressMonitor(new CustomProgressMonitor(repository))
+      val git = Try(command.call())
```

Comment:
I think that the standard try catch is simpler in this case (and couple other in this PR)
```
try command.call()
catch {
  case e: Exception => throw new IllegalStateException(s"Could not clone git $repository: ${e.getMessage}")
}
```